larpandoracontent/LArControlFlow/NeutrinoIdTool.cc

```diff
@@ -156,18 +161,20 @@ void NeutrinoIdTool::Collect2DHits(const PfoList &pfos, CaloHitList &reconstruct
 
     for (const CaloHit *const pCaloHit : collectedHits)
     {
-        if (!reconstructableCaloHitSet.count(pCaloHit))
+        const CaloHit *const pParentHit = static_cast<const CaloHit *>(pCaloHit->GetParentAddress());
+        if (!reconstructableCaloHitSet.count(pParentHit))
```

Comment:
Does this mean that the current implementation isn't quite doing what we thought it was doing? IIRC, the master instance runs this, but it's reviewing particles made by the slice neutrino worker instance or slice cosmic worker instance, which have hits that have been copied into this instance. So, this now looks correct, but the count has always previously returned zero, e.g. for MicroBooNE. Does this match your unstinting @a-d-smith? Would this likely require training update if corrected?

Reply:
I think Ed's approach here is correct.

I've just checked back to the code I initially wrote to about 2 years ago when I did the MicroBooNE training. Back then I was indeed accessing the parent address of the hits as Ed is doing here. So I think the MicroBooNE SVM file is okay. I assume this line must have been changed during the various refactors that have gone on since then. 

This code (tagged release v03_11_00): 
https://github.com/PandoraPFA/LArContent/blob/v03_11_00/larpandoracontent/LArControlFlow/NeutrinoIdTool.cc#L142

...was used to make the current SVM data file:
https://github.com/PandoraPFA/LArMachineLearningData/blob/master/PandoraSvm_v03_11_00.xml